src/main/scala/temple/generate/utils/CodeTerm.scala

```diff
@@ -96,8 +96,22 @@ object CodeTerm {
     def spaced(string: CodeTerm*): String = mkCode(start, "\n", indent(mkCode(string)), "\n", end)
   }
 
-  object codeParens extends codeParens("(", ")") {
-    object curly  extends codeParens("{", "}")
-    object square extends codeParens("[", "]")
+  object codeWrap {
+
+    object parens extends codeWrap("(", ")") {
+      object block extends codeWrap("(\n", ")\n")
+    }
+
+    object curly extends codeWrap("{", "}") {
+      object block extends codeWrap("{\n", "}\n")
+    }
+
+    object square extends codeWrap("[", "]") {
+      object block extends codeWrap("[\n", "]\n")
```

Comment:
```scala
def block: CodeWrap = new CodeWrap(s"$start\n",s"\n$end")
```
On the class, rather than repeating per object. If you don’t want to enable this for quotes, then make a subclass CodeBlockWrap or similar. Then square, curly etc can be made vals rather than objects

Reply:
This already exists, it’s CodeWrap#spaced